src/ssr/src/extractStyles.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import {
+  extractStyles as boxExtractStyles,
+  hydrate as boxHydrate
+} from 'ui-box'
+import { rehydrate } from 'glamor'
+import { renderStatic } from 'glamor/server'
+
+const EVERGREEN_SCRIPT_ID = 'evergreen-hydrate'
+const isDev = process.env.NODE_ENV === 'development' || !process.env.NODE_ENV
+
+/**
+ * You shouldn't have to manually run this.
+ * This is mainly an export for testing purposes.
+ */
+export function hydrate(hydration) {
+  if (hydration.uiBoxCache) {
+    boxHydrate(hydration.uiBoxCache)
+  }
+
+  if (hydration.glamorIds) {
+    rehydrate(hydration.glamorIds)
+  }
+}
+
+function autoHydrate() {
+  if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+    const hydration = document.getElementById(EVERGREEN_SCRIPT_ID)
+
+    if (hydration) {
+      try {
+        const hydrationObject = JSON.parse(hydration.innerHTML)
+        hydrate(hydrationObject)
+      } catch (err) {
+        if (isDev) {
+          console.log(
+            'Evergreen automatic hydration object is invalid JSON',
+            err
+          )
+        }
+      }
+    }
+  }
+}
+
+autoHydrate()
```

Comment:
Unfortunately we'll need to remove `"sideEffects": false` from the `package.json` for this to work correctly in Webpack 4, which will result in tree shaking not working very well. Ideally the user should manually call `autoHydrate()` before rendering and we don't call it here.